README.md

```diff
@@ -35,8 +35,107 @@ Network configuration
 ---
 
 Firecracker, by design, only supports Linux tap devices. The SDK
-provides facilities to attach a tap device to the Firecracker VM, but
-the client is responsible for further configuration.
+provides facilities to:
+* Attach a pre-created tap device, optionally with static IP configuration, to
+  the VM. This is referred to as a "static network interface".
+* Create a tap device via [CNI](https://github.com/containernetworking/cni) plugins, 
+  which will then be attached to the VM automatically by the SDK. This is referred 
+  to as a "CNI network interface"
+  
+### CNI
+If a VM is configured with a CNI network interface, by default CNI configuration
+files will be sought from `/etc/cni/conf.d` and CNI plugins will be sought under
+`/opt/cni/bin` (both of these values can be overridden via API fields).
+
+It's currently highly recommended to use CNI configuration that includes
+[tc-redirect-tap](cni/Makefile) as a chained plugin. This will allow you to
+adapt pre-existing CNI plugins/configuration to a tap device usable by a
+Firecracker VM.
+
+#### Example
+
+With the following file at `/etc/cni/conf.d/fcnet.conflist`:
+```
+{
+  "name": "fcnet",
+  "cniVersion": "0.3.1",
+  "plugins"": [
+    {
+      "type": "ptp",
+      "ipMasq": true,
+      "ipam": {
+        "type": "host-local",
+        "subnet": "192.168.127.0/24",
+        "resolvConf": "/etc/resolv.conf"
+      }
+    },
+    {
+      "type": "tc-redirect-tap"
+    }
+  ]
+}
+```
+
+and the 
+[`ptp`](https://github.com/containernetworking/plugins/tree/master/plugins/main/ptp), 
+[`host-local`](https://github.com/containernetworking/plugins/tree/master/plugins/ipam/host-local) 
+and [`tc-redirect-tap`](cni/Makefile)
+CNI plugin binaries installed under `/opt/cni/bin`, you can specify, in the Go SDK API, 
+a `Machine` with the following `NetworkInterface`:
+```
+{
+  NetworkInterfaces: []firecracker.NetworkInterface{{
+    CNIConfiguration: &firecracker.CNIConfiguration{
+      NetworkName: "fcnet",
+      IfName: "veth0",
+    },
+  }}
+}
+```
+
+Note that `NetworkName` in the `CNIConfiguration` of the API matches the `name` field 
+specified inside the `/etc/cni/conf.d/fcnet.conflist` file.
```

Comment:
Are you required to have the `/etc/cni/conf.d/fcnet.conflist` file?  Is it possible to specify all of these options via an API call instead of requiring an on-disk file?

Reply:
Right now, yes you have to make a file. I thought about what you're suggesting too, but specifying these settings in a file is the most typical way of using CNI and what other comparable frameworks ask for, so I think asking for a filepath is the most basic support we should offer. We can very easily add support for specifying all of the parameters directly via the API on top of this, but that should be a next iteration IMO. I'll open an issue for doing so if there's no disagreement.

Reply:
Sounds good, but can you make it clear that the file is required for now?

Reply:
Sure, in the section above I mention where the SDK looks for the CNI configuration files, I'll add an explicit sentence about it there